src/nix/main.cc

```diff
@@ -356,12 +356,12 @@ void mainWrapped(int argc, char * * argv)
         && args.command->first != "upgrade-nix")
         settings.requireExperimentalFeature(Xp::NixCommand);
 
-    if (args.useNet && !haveInternet()) {
+    if (!settings.offline && !haveInternet()) {
         warn("you don't have Internet access; disabling some network-dependent features");
-        args.useNet = false;
+        settings.offline = false;
```

Comment:
`= true;`, should be opposite of `useNet` and `false` doesn't make sense here anyway since it's only reached if it's already `false`.

Reply:
Good catch!